conbench/tests/entities/test_commit.py

```diff
@@ -62,27 +65,32 @@ def test_get_github_commit():
         "author_name": "Diana Clarke",
         "author_login": "dianaclarke",
         "author_avatar": "https://avatars.githubusercontent.com/u/878798?v=4",
+        "branch": branch,
+        "fork_point_sha": sha,
     }
-    assert get_github_commit(repo, sha) == expected
+    assert get_github_commit(repo, branch=branch, sha=sha) == expected
 
 
 @pytest.mark.slow
-def test_get_github_commit_pull_request():
+def test_get_github_commit_and_fork_point_sha_pull_request():
     # NOTE: This integration test intentionally hits GitHub.
     # TODO: This test will fail once it's no longer one of the most recent 1000
     # commits to the apache/arrow repository.
     repo = "https://github.com/apache/arrow"
     sha = "982023150ccbb06a6f581f6797c017492485b58c"
+    branch = "dianaclarke:ARROW-13266"
     tz = dateutil.tz.tzutc()
     expected = {
-        "parent": "780e95c512d63bbea1e040af0eb44a0bf63c4d72",
+        "parent": "c8668f85a465ea05b2724ec47ff72c4db4d7dfe6",
```

Comment:
This was broken on main. By default we don't `--run-slow` so we weren't testing this.

Note that `c866` is the *true* parent of `9820`, which also happens to be on the same PR branch. `780e` is the fork point of this PR branch.

Reply:
Yeah, I ran the test on main and it failed. So `get_github_commit()` has been returning the true git parent for some time. Looking at the code, that makes sense. 

https://github.com/conbench/conbench/blob/ef88d427dd6be11cdd95e17da4e9940927da37e5/conbench/entities/commit.py#L155-L158

Basically, if the true parent is in the last 1000 commits—which it always is!—it will return the true parent as the parent. So this whole time there has been a bug in the behavior we *didn't* want, resulting in the behavior we wanted.

However, to add to the confusion, when a user hits the API for a history, we execute [this other code](https://github.com/conbench/conbench/blob/ef88d427dd6be11cdd95e17da4e9940927da37e5/conbench/entities/distribution.py#L139) to "get the closest parent" which is actually looking for the fork point. So that code is probably what we were thinking is wrong.

Reply:
Aaaah I see. I'm very glad that we're cleaning up this confusing behavior since it apparently fooled me (and likely the original authors too!) here

Reply:
Me too!! (and I'll actually clean that code up in a subsequent PR; I'm trying to keep this PR to just "add these two fields to the Commit table")